drivers/gpio/gpio_mcp23x17.c

```diff
@@ -130,11 +166,14 @@ static int write_port_regs(const struct device *dev, uint8_t reg,
 
 	ret = spi_write(drv_data->spi, &drv_data->spi_cfg, &tx);
 	if (ret) {
-		LOG_DBG("spi_write FAIL %d\n", ret);
-		return ret;
+		LOG_ERR("MCP23X17[0x%X]: SPI error writing to register 0x%X (%d)", addr, reg, ret);
 	}
+#endif  /* MCP23X17_BUS_SPI */
 
-	return 0;
+	LOG_DBG("MCP23X17: Write: REG[0x%X] = 0x%X, REG[0x%X] = 0x%X",
```

Comment:
OK, I see the debug message moved here.  That's ok, but since it's available please include `ret` as one of the things it displays.

Reply:
Will do that.

Reply:
On the other hand; if the error-message is printed it states the return value there. Otherwise it is 0. 
It's implicit in the debug-line.

Reply:
Assuming neither the debug line nor the error line are dropped by the logger.